pytorch_lightning/profiler/profilers.py

```diff
@@ -416,8 +399,10 @@ def __init__(
             self.on_train_start(local_rank=local_rank)
             self.on_train_start = super().on_train_start
 
-    def on_train_start(self, local_rank: Optional[str] = None):
+    def on_train_start(self, local_rank: Optional[str] = None, dir_path: str = None):
         self.local_rank = local_rank
+        if self.path_to_export_trace is None:
```

Comment:
this seems to be fishy, so you can set it only once?
then add branch with a warning that you passed `dir_path` which wont be used

Reply:
Add a description to explain how it works.